text/0110-z-pages.md

```diff
@@ -0,0 +1,37 @@
+# zPages: general direction (#110)
+
+Make zPages a standard OpenTelemetry component.
+
+## Motivation
+
+Self-introspection debug pages or zPages are in-process web pages that display collected data from the process they are attached to. They are used to provide in-process diagnostics without the need of any backend to examine traces or metrics. Various implementations of zPages are widely used in many environments. The standard extensible implementation of zPages in OpenTelemetry will benefit everybody.
+
+## Explanation
+
+This OTEP is a request to get a general approval for zPages development as an experimental feature [open-telemetry/opentelemetry-specification#62](https://github.com/open-telemetry/opentelemetry-specification/pull/632). See [opencensus.io/zpages](https://opencensus.io/zpages/) for the overview of zPages.
+
+## Internal details
+
+Implementation of zPages includes multiple components - data collection (sampling, filtering, configuration), storage and aggregation, and a framework to expose this data.
+
+This is a request for a general direction approval. There are a few principles for the development:
+
+1. zPages MUST NOT be hardcoded into OpenTelemetry SDK.
```

Comment:
What do you mean by "not be hardcoded"? Something like this?
```suggestion
1. zPages MUST NOT be enabled or configured by default in the OpenTelemetry SDK.
```

Reply:
Or does this mean it must be supplied in a separate package/code unit rather than "weaved into" the SDK?

Reply:
@SergeyKanzhelev Please clarify.

Reply:
I agree with your interpretation about making zPages a separate package / code unit, at the level of a span process (for tracing). In OpenCensus, the zPages support was baked-in to the default SDK tracer. This relates to https://github.com/open-telemetry/opentelemetry-specification/issues/373 about having access to live, unfinished spans.